### PR TITLE
#21 implementation avoid cyclic dependencies

### LINKED ISSUE #21: Rule to avoid cyclic dependencies
With ArchUnit we can add a rule like this:
```
slices()
        .matching("«pattern-to-be-defined»")
        .should()
        .beFreeOfCycles()
        //.ignoreDependency(alwaysTrue(), simpleNameEndingWith("Repository"))
        .because("Cyclic dependencies should be prevented.");
```

The ignore is just some thought as example if e.g. two components both call their own repository and the foreign one.
While I still think this should be architecture validation but if for some reasons component cut was designed wrong, one should have some hint how to deal with this. However, we should most probably address such excuses via some own documentation with deep-links to ArchUnit exclusion/ignore features.

### PR DESCRIPTION
**Pull request by NicolasVanBellen**

### CODE REVIEW

**Review by hohwille — COMMENTED** [review]
@NicolasVanBellen Thanks for your PR.
This is looking very good and seem functional.
However, I do have a question. Can you please comment.
If my guess does not work, we can still merge and rework later.

File: src/test/java/com/devonfw/sample/archunit/AvoidCyclicDependenciesTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class AvoidCyclicDependenciesTest {
+    @ArchTest
+    static final ArchRule no_cyclic_dependencies_are_allowed =
+    slices()
+        .matching("..archunit.(*)..")
```
Comment (hohwille) [inline src/test/java/com/devonfw/sample/archunit/AvoidCyclicDependenciesTest.java]: here we encode our project specific namespace into the rule.
The first thing that projects will have to do after refactoring this sample to their own project namespace is to identify all these places and change them according to their project.
With `StructuredPackage` we have created a solution that is fully generic and should apply to all devonfw based Java projects no matter what their package namespace actually is.
I do not know how this ArchUnit patterns are working. Are they Java regular expressions?
As `..` behaves differently from what I know about regex this looks more special.
However, I would simply try something like this instead:
```
  .matching("..(*).(common|service|logic|dataaccess|batch|client")..")
```

Comment (NicolasVanBellen) [reply src/test/java/com/devonfw/sample/archunit/AvoidCyclicDependenciesTest.java]: Your approach does work. I have adjusted it in the file and we will also adjust it in #29 as well.

**Review by hohwille — APPROVED** [review]
@NicolasVanBellen thanks for testing and rework. 👍 
Now ready for merge.

**Pull request merged**